airflow/operators/python.py

```diff
@@ -547,14 +547,21 @@ def __init__(
     ):
         if (
             python_version
-            and str(python_version)[0] != str(sys.version_info.major)
+            and python_version[0] != str(sys.version_info.major)
             and (op_args or op_kwargs)
         ):
             raise AirflowException(
                 "Passing op_args or op_kwargs is not supported across different Python "
                 "major versions for PythonVirtualenvOperator. Please use string_args."
                 f"Sys version: {sys.version_info}. Venv version: {python_version}"
             )
+        if type(python_version) is float or type(python_version) is int:
+            raise TypeError(
+                "Passing numeric type(int, float) as python_version "
+                "has been removed for PythonVirtualenvOperator. "
+                "Since Python 3.10 is interpreted as 3.1. "
+                "Please use string value instead."
+            )
```

Comment:
This is breaking changes for someone who use python 3.8 and 3.9.
I think warning would enough, if we can't resolve `3.10` vs `3.1`

Reply:
I think users can't find that python is running with 3.1 if there's no error.
Then, should I change it to print just warning message?

Reply:
I guess it is edge case with Python 3.1, I'm not sure that something below 3.3 in 3.x series would ever been stable for production usage. So warning would be enough.

Reply:
changed it to warning. thank you. :)